src/porting.cpp

```diff
@@ -798,5 +800,36 @@ v2u32 getDisplaySize()
 #	endif // __ANDROID__
 #endif // SERVER
 
+
+#ifdef WIN32
+
+bool secure_rand_fill_buff(unsigned char *buf, size_t buf_size)
+{
+	HCRYPTPROV wctx;
+
+	CryptAcquireContext(&wctx, NULL, NULL, PROV_RSA_FULL, CRYPT_VERIFYCONTEXT);
+	CryptGenRandom(wctx, buf_size, (BYTE *)buf);
+	CryptReleaseContext(wctx, 0);
+	return true;
+}
+
+#else
+
+bool secure_rand_fill_buff(unsigned char *buf, size_t buf_size)
+{
+
+	FILE *fp = fopen("/dev/urandom", "r");
+
+	if (fp) {
+		fread(buf, buf_size, 1, fp);
+		fclose(fp);
+		return true;
+	} else {
+		return false;
+	}
+}
+
+#endif
+
```

Comment:
I don't really like functions that return a value not having a return value at the end of the function (even though it's never reached). Some compilers will also issue a warning when you construct the function in this manner. Perhaps the code below is "better".

bool secure_rand_fill_buff(unsigned char *buf, size_t buf_size)
{

	FILE *fp = fopen("/dev/urandom", "r");

	if (!fp)
		return false;

	fread(buf, buf_size, 1, fp); // Perhaps the value fread() returns should be checked?
	fclose(fp);
	return true;
}